### ISSUE TITLE
Support Proxy Configuration

### ISSUE THREAD

**Issue by joramhoefs** [8p/EightPointsGuzzleBundle#97]
The current implementation of the ConfigurationInterface doesn't support the proxy client configuration. So i would propose the following addition to the _options_ arrayNode:

```
->arrayNode('proxy')
    ->children()  
        ->scalarNode('http')->end()  
        ->scalarNode('https')->end()  
        ->scalarNode('no')->end()  
    ->end()  
->end()
```

**Comment by heddn** [comment]
It actually seems to drop support for a lot of things. I feel like it should support all of http://docs.guzzlephp.org/en/latest/request-options.html.

**Comment by heddn** [comment]
I'm wanting to add http://docs.guzzlephp.org/en/latest/request-options.html#allow-redirects.

**Comment by gregurco** [comment]
Work in progress :slightly_smiling_face: I will try to add support as many options as possible.

**Comment by gregurco** [comment]
This issue was fixed in #107 and #140. I close the issue.
